# 'likes' Report

### By Serena Gestring

#### September 20, 2022



The data displayed below was collected by the company Instagram. This data was collected to be used for three main purposes: content, marketing, and research. Firstly, companies like Instagram use this data to determine what content to provide the user. For example, the data could show that a user likes a number of posts from certain accounts, or the consistently likes content relating to specific interests, such as a sport or hobby. Now Instagram knows to provide more of that content in hopes the user will continue engaging with Instagram. This is essentially using data on past engagement to shape future engagement. Similarly, companies also use this data to determine what ads to show the user while on the platform. Data has made targeting specific audiences much easier because it allows companies to show each individual user ads unique to them and their interests. Finally, this data is useful for researchers. Humanity has never had access to this much global data at any other point in history. This data can tell us a lot about ourselves, and allow researchers to predict specific traits of individual users. 

For data to be reliable, it needs to be accurate and complete. This data is reliable in that is it valid, meaning it has been formatted and stored in a way that allows for organization and analysis. The Instagram data below is organized in a pandas DataFrame consisting of two columns: one providing the name of the account associated with the liked post, and the other providing the date the post was liked. This data was provided to the user by Instagram, and because Instagram is the keeper of the data, the company can decide what to share with the user and what not to share. The data could be incomplete, which would mean it does not actually provide an accurate depiction of reality, and therefore we cannot draw reliable or accurate conclusions from it in order to make truly informed decisions.   

Research and data analysis is important for social media companies like Instagram to keep their users engaged with their product. This process involves developing and testing hypotheses. A theoretical hypothesis could be something like, "If we increase the amount of content from these types of accounts, then the user's engagement with the app will also increase." A statistical hypothesis could be something like, "Increasing the amount of content from certain types of accounts by [_]% will result in an increase in user engagement by [_]%." The data file below would provide the "before" data, similar to the control of an experiment, essentially giving a baseline by which we can compare the data gathered after the change in content. If Instagram's main goal is to increase user engagement, this is one way the company can try to do that. Instagram could also send out a survey to its users to get qualitative data on what the users claim would motivate them to engage more with the app. 

In [1]:
import pandas as pd
import json

In [2]:
with open(r"C:\Users\Serena\EMAT22110_FA22\technogecko_20200714_toshare\technogecko_20200714_toshare\likes.json") as l:
    likes_data = json.load(l)

In [ ]:
likes_data

In [3]:
likes_data_df = pd.DataFrame()

In [4]:
likes = pd.DataFrame(likes_data['media_likes'])

In [5]:
likes

0                 1
0    2020-07-11T04:39:28+00:00   ball_doesnt_lie
1    2020-07-11T04:39:05+00:00   ball_doesnt_lie
2    2020-07-05T17:25:44+00:00     ali_saurusrex
3    2020-07-03T03:40:02+00:00  cacandassociates
4    2020-06-25T17:41:50+00:00  cacandassociates
..                         ...               ...
330  2013-02-05T02:58:46+00:00            natgeo
331  2013-02-05T02:22:24+00:00       aroseroar16
332  2013-02-05T01:29:31+00:00       aroseroar16
333  2013-02-04T17:42:04+00:00     ali_saurusrex
334  2013-02-04T17:39:15+00:00      a_matt_silva

[335 rows x 2 columns]

In [6]:
likes.groupby(1).count()

0
1                              
a_matt_silva                 18
ali_saurusrex                68
andrewtcurry                  1
aroseroar16                  29
athenalemon                   1
ball_doesnt_lie               2
bianchi_usa                   2
bsnder                        5
cacandassociates             15
camecol501                    2
carswithoutlimits             1
catsandkilos                  1
cbarker13                     9
colin_storm                  11
danneabreanne                13
drchadlarson                 10
emmyr0o                      16
favoritedekruyfonthecitadel   4
holly.berry_22                1
ibiscycles                    1
inalull                      10
jackie.cc                     9
jennjohnn                     3
jessicaajeean                 2
jhushlordofthesith            1
johnnydhampton                1
jsevier                       1
kateblarson                   2
kellyc00k                     1
leleebabe                     3
lewkmiller                    2
melissarohwer                 1
meredith.wang                 9
miles_sari                    3
mtb_crohnert92                5
mycaninelife                  2
natgeo                        1
odonnellnic                   1
orangekoala2                 27
phd_comics                    1
poorlydrawnlines              2
priscillaannnnn               3
reams_esq                    16
rockymountainbicycles         1
rskinner90                    1
samrosehill                   2
samuel.rhodes                 2
shaunacoxsey                  1
silva_c097                    3
slemdawgmillionaire           1
sp4zie                        1
theradavist                   1
thesignpad                    1
transitionbikes               1
tslemmer                      2
vitalmtb                      1
wrenchscienceusa              1

The following are the top five accounts with the highest number of likes (provided by the table above):
1. The account "ali_saurusrex" has the highest number of likes with a total of 68 liked posts. It should be noted that this is considerably higher than the number of likes for any other account provided in the data, including the account with the second highest number of likes. One might even consider this an outlier. 
2. The account "aroseroar16" has the second highest number of likes with a total of 29 liked posts. Again, note that this is a considerable distance away from the most-liked account.
3. The account with the third highest number of likes is "orangekoala2" with a total of 27 liked posts. 
4. The account with the fourth highest number of likes is "a_matt_silva" with a total of 18 liked posts.
5. The account "emmyr0o" and the account "reams_esq" both have the fifth highest number of likes with a total of 16 liked posts each. 

The remaining accounts listed in the DataFrame have less than 16 liked posts. 

As can be seen in the DataFrame, from 2013 to 2020, this Instagram user only liked 335 posts total. This means that in a seven year timespan, this user liked an average of ~48 posts per year, or an average of ~4 posts per month. While this is not an indicating of how frequently the user was actually using Instagram app, it is an indication of how much the user was *interacting* with Instagram content (liking posts). While this is the only empirical data being provided in this report, it can be reasonably assumed that there are Instagram users who interact with the app more than this user (as well as users who interact even less). Therefore, Instagram can use this what the user has liked in the past to predict what they will like in the future. Looking at the accounts whose posts the user liked the most (listed above), Instagram can feed the user more content from those accounts or content that is similar enough that it is probable the user will like that content as well. This strategy could spur increased engagement on the part of the user, as stated in the hypothesis. Again, this data would have to be compared to data gathered later after the content change in order to make any real conclusions regarding that hypothesis. 

This data is insightful for learning about this individual user, but provides a very limited view of the bigger picture one its own. A potential next step for Instagram would be to compare this data with the data of thousands of other users to look for similarities and patterns. This research will help Instagram make decisions regarding development of its user experience and any other goals the company is working towards.
